### ISSUE TITLE
Installation failed (The system requirements are not satisfied.)

### ISSUE THREAD

**Issue by MasterOfStar** [intel/haxm#387]
PowerShell:
>\> ./checktool.exe --verbose
CPU vendor          *  GenuineIntel
Intel64 supported   *  Yes
VMX supported       *  Yes
VMX enabled         *  Yes
EPT supported       *  Yes
NX supported        *  Yes
NX enabled          *  Yes
Hyper-V disabled    *  Yes
OS version          *  Windows 10.0.19042
OS architecture     *  x86_64
Guest unoccupied    *  Yes. 0 guest(s)
————————————————

LOG:
>Version: 7.6.5
Output folder: C:\Program Files\Intel\HAXM
Extract: checktool.exe... 100%
Execute: C:\Program Files\Intel\HAXM\checktool.exe --verbose
The system requirements are not satisfied.

It is look like compliant [Configurations](https://github.com/intel/haxm/wiki/Windows-Installation-Environment-Check-and-Configurations) but cannot be installed

**Comment by wcwang** [comment]
Thanks for your report. Could you try the latest release HAXM v7.7.0 instead? And then find the latest log that prompted in the wizard dialog. Thanks.

**Comment by MasterOfStar** [comment]
> Thanks for your report. Could you try the latest release HAXM v7.7.0 instead? And then find the latest log that prompted in the wizard dialog. Thanks.

Version: 7.7.0
Output folder: C:\Program Files\Intel\HAXM
Extract: checktool.exe... 100%
Execute: C:\Program Files\Intel\HAXM\checktool.exe --verbose
The system requirements are not satisfied.

**Comment by MateuszLas421** [comment]
check Hyper-V  , A windows update turned on Hyper-V . Help me  ` > bcdedit /set hypervisorlaunchtype off`

**Comment by wayne-ma** [comment]
Hi @MasterOfStar ,what is your CPU model?

**Comment by MasterOfStar** [comment]
> Hi @MasterOfStar ,what is your CPU model?

Intel Core i7-9750H

**Comment by wcwang** [comment]
> Version: 7.7.0
> Output folder: C:\Program Files\Intel\HAXM
> Extract: checktool.exe... 100%
> Execute: C:\Program Files\Intel\HAXM\checktool.exe --verbose
> The system requirements are not satisfied.

Thanks for your information. The weird phenomenon in your log is there is no detailed output after checktool.exe execution. But it seems that you are able to acquire the correct checking result when you run check tool independently. The log of HAXM v7.7.0 came from %TEMP% folder, right? Have you ever installed HAXM successfully?

**Comment by MasterOfStar** [comment]
> > Version: 7.7.0
> > Output folder: C:\Program Files\Intel\HAXM
> > Extract: checktool.exe... 100%
> > Execute: C:\Program Files\Intel\HAXM\checktool.exe --verbose
> > The system requirements are not satisfied.
> 
> Thanks for your information. The weird phenomenon in your log is there is no detailed output after checktool.exe execution. But it seems that you are able to acquire the correct checking result when you run check tool independently. The log of HAXM v7.7.0 came from %TEMP% folder, right? Have you ever installed HAXM successfully?

Yes, log file from %TEMP%. And I didn't succeed in installing HAXM.
I gave up installing HAXM and used a physical Android phone to debug the apk.

**Comment by wcwang** [comment]
Sorry for bringing you inconvenience. Would you help us to locate this issue so that we can improve the installer program? Could you find the file C:\Program Files\Intel\HAXM\checktool.exe and run below command? If this file doesn't exist, could you check your anti-virus software installed in your Windows host to see whether it was blocked due to the security reason?

`C:\Program Files\Intel\HAXM\checktool.exe --verbose`

**Comment by MasterOfStar** [comment]
> Sorry for bringing you inconvenience. Would you help us to locate this issue so that we can improve the installer program? Could you find the file C:\Program Files\Intel\HAXM\checktool.exe and run below command? If this file doesn't exist, could you check your anti-virus software installed in your Windows host to see whether it was blocked due to the security reason?
> 
> `C:\Program Files\Intel\HAXM\checktool.exe --verbose`

C:\Users\\*MyUsername*\>C:\"Program Files"\Intel\HAXM\checktool.exe --verbose
CPU vendor          *  GenuineIntel
Intel64 supported   *  Yes
VMX supported       *  Yes
VMX enabled         *  Yes
EPT supported       *  Yes
NX supported        *  Yes
NX enabled          *  Yes
Hyper-V disabled    *  Yes
OS version          *  Windows 10.0.19042
OS architecture     *  x86_64
Guest unoccupied    *  Yes. 0 guest(s)

**Comment by wcwang** [comment]
Thanks for your reply. Could you run below command to see whether the output is 0 after running checktool.exe?

`echo %ERRORLEVEL%`

If the output is not 0, try to run below command to disable Hyper-V and reboot your Windows.

`bcdedit /set hypervisorlaunchtype off`

**Comment by MasterOfStar** [comment]
> Thanks for your reply. Could you run below command to see whether the output is 0 after running checktool.exe?
> 
> `echo %ERRORLEVEL%`
> 
> If the output is not 0, try to run below command to disable Hyper-V and reboot your Windows.
> 
> `bcdedit /set hypervisorlaunchtype off`

```
C:\Users\MyUsername>C:\"Program Files"\Intel\HAXM\checktool.exe --verbose
CPU vendor          *  GenuineIntel
Intel64 supported   *  Yes
VMX supported       *  Yes
VMX enabled         *  Yes
EPT supported       *  Yes
NX supported        *  Yes
NX enabled          *  Yes
Hyper-V disabled    *  Yes
OS version          *  Windows 10.0.19042
OS architecture     *  x86_64
Guest unoccupied    *  Yes. 0 guest(s)

C:\Users\MyUsername>echo %ERRORLEVEL%
0
```

**Comment by MasterOfStar** [comment]
And here is a different result.
```
C:\Users\MyUsernam\Downloads\haxm-windows_v7_7_0>haxm-7.7.0-setup.exe
(run in GUI and tell me ‘The system requirements are not satisfied.’)
C:\Users\MyUsername\Downloads\haxm-windows_v7_7_0>echo %ERRORLEVEL%
9009
```

After I ran `bcdedit /set hypervisorlaunchtype off` and restarted:
```
C:\Users\MyUsername>C:\"Program Files"\Intel\HAXM\checktool.exe --verbose
CPU vendor          *  GenuineIntel
Intel64 supported   *  Yes
VMX supported       *  Yes
VMX enabled         *  Yes
EPT supported       *  Yes
NX supported        *  Yes
NX enabled          *  Yes
Hyper-V disabled    *  Yes
OS version          *  Windows 10.0.19042
OS architecture     *  x86_64
Guest unoccupied    *  Yes. 0 guest(s)

C:\Users\MyUsername>echo %ERRORLEVEL%
0
```
```
C:\Users\MyUsername\Downloads\haxm-windows_v7_7_0>haxm-7.7.0-setup.exe
(run in GUI and tell me ‘The system requirements are not satisfied.’)
C:\Users\MyUsername\Downloads\haxm-windows_v7_7_0>echo %ERRORLEVEL%
9009
```

And when I run haxm-setup and checktool in same cmd.exe:
```
C:\Users\MyUsername>C:\"Program Files"\Intel\HAXM\checktool.exe --verbose
CPU vendor          *  GenuineIntel
Intel64 supported   *  Yes
VMX supported       *  Yes
VMX enabled         *  Yes
EPT supported       *  Yes
NX supported        *  Yes
NX enabled          *  Yes
Hyper-V disabled    *  Yes
OS version          *  Windows 10.0.19042
OS architecture     *  x86_64
Guest unoccupied    *  Yes. 0 guest(s)

C:\Users\MyUsername>echo %ERRORLEVEL%
0

C:\Users\MyUsername>cd .\Downloads\haxm-windows_v7_7_0

C:\Users\MyUsername\Downloads\haxm-windows_v7_7_0>haxm-7.7.0-setup.exe
(run in GUI and tell me ‘The system requirements are not satisfied.’)
C:\Users\MyUsername\Downloads\haxm-windows_v7_7_0>echo %ERRORLEVEL%
0
```

**Comment by wcwang** [comment]
Thanks for your detailed information. That's helpful.

If you have received the error 9009 on your PC, it means that there was a malfunction in your host OS. Can you find another Windows PC to try to install Android Studio or HAXM? Otherwise, perhaps you have to consider checking or re-installing your Windows OS.

**Comment by shyed2001** [comment]
PowerShell 7.1.3
Copyright (c) Microsoft Corporation.

https://aka.ms/powershell
Type 'help' to get help.

PS C:\Users\shyed> C:\"Program Files"\Intel\HAXM\checktool.exe --verbose
CPU vendor          *  GenuineIntel
Intel64 supported   *  Yes
VMX supported       *  Yes
VMX enabled         *  Yes
EPT supported       *  Yes
NX supported        *  Yes
NX enabled          *  Yes
Hyper-V disabled    *  Yes
OS version          *  Windows 10.0.19043
OS architecture     *  x86_64
Guest unoccupied    *  Yes. 0 guest(s)
PS C:\Users\shyed> echo %ERRORLEVEL%
%ERRORLEVEL%
PS C:\Users\shyed> bcdedit /set hypervisorlaunchtype off
The operation completed successfully.
PS C:\Users\shyed> echo %ERRORLEVEL%
%ERRORLEVEL%
PS C:\Users\shyed> C:\"Program Files"\Intel\HAXM\checktool.exe --verbose
CPU vendor          *  GenuineIntel
Intel64 supported   *  Yes
VMX supported       *  Yes
VMX enabled         *  Yes
EPT supported       *  Yes
NX supported        *  Yes
NX enabled          *  Yes
Hyper-V disabled    *  Yes
OS version          *  Windows 10.0.19043
OS architecture     *  x86_64
Guest unoccupied    *  Yes. 0 guest(s)
PS C:\Users\shyed> echo %ERRORLEVEL%
%ERRORLEVEL%
PS C:\Users\shyed>

**Comment by shyed2001** [comment]
> Thanks for your reply. Could you run below command to see whether the output is 0 after running checktool.exe?
> 
> `echo %ERRORLEVEL%`
> 
> If the output is not 0, try to run below command to disable Hyper-V and reboot your Windows.
> 
> `bcdedit /set hypervisorlaunchtype off`


This helped me .
=====

**Comment by MasterOfStar** [comment]
Interesting. For the past few months I've been debugging with a real Android Phone.
Today when I installed this program on a whim,It's install **successful.**
@wcwang tell me, it's
>it means that there was a malfunction in your host OS  ...... you have to consider checking or re-installing your Windows OS

But ... I didn't do anything, Just normal system updates and used.
I can only attribute this to a metaphysical problem. 😂

**Comment by wayne-ma** [comment]
Good to know you could install HAXM successfully now. Seems the hyper-V status blocked your installation before since 'bcdedit /set hypervisorlaunchtype off' resolved it.

**Comment by MasterOfStar** [comment]
> Good to know you could install HAXM successfully now. Seems the hyper-V status blocked your installation before since 'bcdedit /set hypervisorlaunchtype off' resolved it.

no... I think not.
Because I set off hyper-v earlier than type that command.
And when I used that command it didn't install successfully either.
It seems windows BUG for Hyper-V status. You close it, and it shows closed, but the hidden parameters may not be closed properly.

**Comment by AshitaKaze** [comment]
same problem.
----------------------
C:\Program Files\Intel\HAXM>checktool.exe --verbose
CPU vendor          *  GenuineIntel
Intel64 supported   *  Yes
VMX supported       *  Yes
VMX enabled         -  No
EPT supported       *  Yes
NX supported        *  Yes
NX enabled          *  Yes
Hyper-V disabled    *  Yes
OS version          *  Windows 10.0.19042
OS architecture     *  x86_64
Guest unoccupied    *  Yes. 0 guest(s)
-------------------------------
 C:\Program Files\Intel\HAXM>echo %ERRORLEVEL%
1
C:\Program Files\Intel\HAXM>bcdedit /set hypervisorlaunchtype off
access is denied.

**Comment by wcwang** [comment]
@AshitaKaze, you have to run the command in an elevated Command Prompt and enable VMX from BIOS settings. Thanks.

**Comment by AshitaKaze** [comment]
> @AshitaKaze, you have to run the command in an elevated Command Prompt and enable VMX from BIOS settings. Thanks.

Yes, I found that virtualization was not turned on through another issue

**Comment by himanshu-sep** [comment]
Version: 7.6.5
Output folder: C:\Program Files\Intel\HAXM
Extract: checktool.exe... 100%
Execute: C:\Program Files\Intel\HAXM\checktool.exe --verbose
The system requirements are not satisfied.
